nimble/host/services/audio/bass/include/services/bass/ble_audio_svc_bass.h

```diff
@@ -0,0 +1,174 @@
+#define BLE_SVC_AUDIO_BASS_CTRL_POINT_EVENT_REMOTE_SCAN_STARTED             0x01
+#define BLE_SVC_AUDIO_BASS_CTRL_POINT_EVENT_ADD_SOURCE                      0x02
+#define BLE_SVC_AUDIO_BASS_CTRL_POINT_EVENT_MODIFY_SOURCE                   0x03
+#define BLE_SVC_AUDIO_BASS_CTRL_POINT_EVENT_SET_BROADCAST_CODE              0x04
+#define BLE_SVC_AUDIO_BASS_CTRL_POINT_EVENT_REMOVE_SOURCE                   0x05
+
+enum ble_svc_audio_bass_big_enc {
+    BLE_SVC_AUDIO_BASS_BIG_ENC_NOT_ENCRYPTED,
+    BLE_SVC_AUDIO_BASS_BIG_ENC_BROADCAST_CODE_REQ,
+    BLE_SVC_AUDIO_BASS_BIG_ENC_DECRYPTING,
+    BLE_SVC_AUDIO_BASS_BIG_ENC_BAD_CODE
+};
+
+enum ble_svc_audio_bass_pa_sync {
+    BLE_SVC_AUDIO_BASS_PA_SYNC_DO_NOT_SYNC,
+    BLE_SVC_AUDIO_BASS_PA_SYNC_SYNC_PAST_AVAILABLE,
+    BLE_SVC_AUDIO_BASS_PA_SYNC_SYNC_PAST_NOT_AVAILABLE
+};
+
+enum ble_svc_audio_bass_pa_sync_state {
+    BLE_SVC_AUDIO_BASS_PA_SYNC_STATE_NOT_SYNCED,
+    BLE_SVC_AUDIO_BASS_PA_SYNC_STATE_SYNC_INFO_REQ,
+    BLE_SVC_AUDIO_BASS_PA_SYNC_STATE_SYNCED,
+    BLE_SVC_AUDIO_BASS_PA_SYNC_STATE_SYNCED_FAILED,
+    BLE_SVC_AUDIO_BASS_PA_SYNC_STATE_NO_PAST
+};
+
+struct ble_svc_audio_bass_subgroup {
+    uint32_t bis_sync_state;
+    uint8_t metadata_length;
+    uint8_t *metadata;
+};
+
+struct ble_svc_audio_bass_receiver_state_params {
+    ble_addr_t source_addr;
+    uint8_t source_adv_sid;
+    uint32_t broadcast_id;
+    uint8_t pa_sync_state;
+    uint8_t big_encryption;
+    uint8_t bad_code[BLE_AUDIO_BROADCAST_CODE_SIZE];
+    uint8_t num_subgroups;
+    struct ble_svc_audio_bass_subgroup
+        subgroups[MYNEWT_VAL(BLE_SVC_AUDIO_BASS_MAX_SUBGROUPS_PER_RECV_STATE)];
+};
+
+struct ble_svc_audio_bass_receiver_state {
+    uint8_t source_id;
+    struct ble_audio_bsnk *bsnk;
+    struct ble_svc_audio_bass_receiver_state_params params;
+};
+
+struct ble_svc_audio_bass_ctrl_point_event {
+    /**
+     * Indicates the type of BASS event that occurred.  This is one of the
+     * BLE_SVC_AUDIO_BASS_CTRL_POINT_EVENT codes.
+     */
+    uint8_t op;
+
+    /**
+     * A discriminated union containing additional details concerning the BASS Control Point
+     * event.  The 'type' field indicates which member of the union is valid.
+     */
+    union {
+        /**
+         * Represents Add Source operation.  Valid for the following event
+         * types:
+         *     o BLE_SVC_AUDIO_BASS_CTRL_POINT_EVENT_ADD_SOURCE
+         * Application can accept or reject Add Source operation. If no application callback is set
+         * and free Receive State characteristic exists operation is automatically accepted.
+         * If application callback exists and returns 0 accepted. Otherwise, Operation is rejected.
+         * If operation is accepted by application callback, this callback may select receiver
+         * state to be filled. If application doesnt select characteristic, BASS Server falls back
+         * to searching free one. If none is found, operation is rejected.
+         */
```

Comment:
Once the Add Source operation is accepted, the additional event shall be generated to notify all subscribers that Broadcast Source has been added.
As discussed offline, the ble_audio_event should be extended with events that inform that:
* Broadcast Source has been added
* Broadcast Source has been removed
* Broadcast Source has been modified
* Broadcast Code has been provided